lib.js

```diff
@@ -0,0 +1,215 @@
+  lastCid = new URL(contentUrl).host;
+
+  window.localStorage.lastCid = lastCid;
+  console.log(contentUrl);
+
+  return lastCid;
+
+}
+
+export async function removeFile(filename){
+
+  let lastCid = window.localStorage.getItem('lastCid');
+
+  const _removeFile = async ( ) => {
+    let url = `ipfs://${lastCid?lastCid:''}/ipmb-db/${filename}`
+    console.log(`DELETING ${url}`);
+    let response = await fetch(url, {
+      method: 'DELETE',
+      mode: 'cors'
+    });
+    let contentUrl = await response.text()
+    let newCid = new URL(contentUrl).host;
+    return newCid;
+  }
+
+  lastCid = await thisIsBlog(_removeFile);
+  return lastCid;
+
+}
+
+// Adds a post and regenerates the index
+export async function postAdd(file, filename){
+
+  let lastCid = window.localStorage.getItem('lastCid');
+
+  const _addFile = async () => {
+    let url = `ipfs://${lastCid?lastCid:''}/ipmb-db/${filename}`
+    console.log(`ADDING ${url}`);
+    let response = await fetch(url, {
+      method: 'POST',
+      body: file,
+      mode: 'cors'
+    });
+    let contentUrl = await response.text()
+    return new URL(contentUrl).host;
+  }
+
+  lastCid = await thisIsBlog(_addFile);
+  return lastCid;
+}
+
+export async function postUpdate(file, filename, originalFilename){
+  let lastCid = window.localStorage.getItem('lastCid');
+
+  const _removeFile = async cid => {
+    let url = `ipfs://${cid}/ipmb-db/${originalFilename}`
+    console.log(`DELETING ${url}`);
+    let response = await fetch(url, {
+      method: 'DELETE',
+      mode: 'cors'
+    });
+    let contentUrl = await response.text()
+    let newCid = new URL(contentUrl).host;
+    return newCid;
+  }
+
+  const _addFile = async cid => {
+    let url = `ipfs://${cid?cid:''}/ipmb-db/${filename}`
+    console.log(`ADDING ${url}`);
+    let response = await fetch(url, {
+      method: 'POST',
+      body: file,
+      mode: 'cors'
+    });
+    let contentUrl = await response.text()
+    return new URL(contentUrl).host;
+  }
+
+  const _chain = async () => {
+    let cid = await _removeFile(lastCid);
+    return await _addFile(cid);
+  }
+
+  lastCid = await thisIsBlog(_chain);
+  return lastCid;
+}
+
+// Add media
+export async function mediaAdd(file){
+  let lastCid = window.localStorage.getItem('lastCid');
+  let url = `ipfs://${lastCid?lastCid:''}/media/${file.name}`
+  console.log(`ADDING ${url}`);
+  let response = await fetch(url, {
+    method: 'POST',
+    body: file,
+    mode: 'cors'
+  });
+  let contentUrl = await response.text()
+  window.localStorage.lastCid = new URL(contentUrl).host;
+  return window.localStorage.lastCid;
+}
+
+// list files in dir
+// TODO remove hard-coded /ipmb-db and pass as parameter
+async function _fetchFolder(cid){
+  let r = await fetch(`ipfs://${cid}/ipmb-db/`, {
+    headers: {
+      'X-Resolve': 'none',
+    }
+  });
+  let d = await r.json();
+  d = d.filter( e => !!e); // empty dir returns `[ null ]`
+
+  let files = [];
+  for (let filename of d){
```

Comment:
One trick I like to use is `Promise.all` paired with `array.map()` so you can do a bunch of stuff in parallel and speed things up significantly.

e.g.

```javascript
const files = await Promise.all(d.map(async (filename) => {
   const something = await whatever()
   return something;
}))
```

Reply:
Yeah, good idea. At some stage we'll need to worry about blogs having many files and batch requests to avoid overwhelming the local node with too many requests at once or just add a random delay before the network request.